utils/codepointtrie/src/codepointtrie.rs

```diff
@@ -383,6 +420,244 @@ impl<'trie, T: TrieValue + Into<u32>> CodePointTrie<'trie, T> {
     pub fn get_u32(&self, code_point: u32) -> u32 {
         self.get(code_point).into()
     }
+
+    /// Returns a [`CodePointMapRange`] struct representing a range of code
+    /// points associated with the same value.
+    pub fn get_range(&self, start: u32) -> Option<CodePointMapRange> {
```

Comment:
*In addition to* this low-level function (especially once the ValueFilter parameter is added), I suggest you also add a Rust-y iterator (and implement it via get_range()), like in Java:

https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/util/CodePointMap.html#iterator--

Reply:
The return type of [`get_script_extensions_val`](https://unicode-org.github.io/icu4x-docs/doc/icu_properties/script/struct.ScriptExtensions.html#method.get_script_extensions_val) is a [ZeroSlice](https://unicode-org.github.io/icu4x-docs/doc/zerovec/struct.ZeroSlice.html). More specifically, a `ZeroSlice<Script>`.  As with everything else named Zero*, it supports a byte array of un-/aligned data on BE/LE endianness, and iteration without incurring allocation.

Above in this PR, in the implementation of `has_script()`, you can see where I call [`ZeroSlice::iter()`](https://unicode-org.github.io/icu4x-docs/doc/zerovec/struct.ZeroSlice.html#method.iter) on that Script_Extensions return value to give me the iterator of Script values.

Reply:
Misunderstanding. I didn't mean an iterator over the scripts for one code point. I meant something more like what Shane sketched out elsewhere, a language-idiomatic iterator over the ranges. Not instead of get_range(), but in addition.